src/analysis/AppLauncher.ts

```diff
@@ -218,11 +251,52 @@ const ApplicationLauncher = _.flow(
     });
 
     useEffect(() => {
-      const runtime = getCurrentRuntime(runtimes);
       const runtimeStatus = getConvertedRuntimeStatus(runtime);
 
+      const setupWelder = _.flow(
+        Utils.withBusyState(setBusy),
+        withErrorReporting('Error setting up analysis file syncing')
+      )(async (runtime: Runtime) => {
+        // The special case here is because for GCP, Jupyter and JupyterLab can both be run on the same runtime and a
+        // user may toggle back and forth between them. In order to keep notebooks tidy and in a predictable location on
+        // disk, we mirror the localBaseDirectory used by edit mode for Jupyter.
```

Comment:
FYI, I do not think this works very well. I always get a notebook not found error when toggling back and forth between the two and have to manually navigate to the file

Reply:
Not saying this is in the scope of this PR or this should be fixed given that it is behind a feature flag but curious to see if you experience the same

Reply:
huh, interesting. I don't toggle back frequently